### PR TITLE
[WIP] Strategies: Implement AAVE v3

### PR DESCRIPTION
**Pull request by facuspagnuolo**
Let's wait until we have tested it using a polygon mainnet fork

### CODE REVIEW

File: packages/strategies/contracts/aave-v3/AaveV3Strategy.sol [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+     */
+    function lastValue(address account) public view override returns (uint256) {
+        return aToken.balanceOf(account);
+    }
+
+    /**
+     * @dev Claims AAVE rewards.
+     */
+    function claim(bytes memory) external override returns (address[] memory tokens, uint256[] memory amounts) {
+        address[] memory aTokens = new address[](1);
```
Comment (dmf7z) [inline packages/strategies/contracts/aave-v3/AaveV3Strategy.sol:93]: Is this the same as doing:

`address[1] memory aTokens;`

File: packages/strategies/contracts/aave-v3/AaveV3Strategy.sol [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    /**
+     * @dev Withdraw tokens from the AAVE lending pool
+     * @param ratio Ratio of the invested position to withdraw
+     * @return amount Amount of strategy tokens exited with
+     * @return value Value represented by the exited amount
+     */
+    function exit(uint256 ratio, uint256, bytes memory) external override returns (uint256 amount, uint256 value) {
+        require(ratio <= FixedPoint.ONE, 'AAVE_V3_INVALID_RATIO');
```
Comment (dmf7z) [inline packages/strategies/contracts/aave-v3/AaveV3Strategy.sol:121]: Do we check at the wallet that ratio > 0? If not we can add here

`if (ratio == 0) return 0;`

**Review by dmf7z — COMMENTED** [review]
👍

File: packages/strategies/contracts/aave-v3/AaveV3Strategy.sol (hunk 1 above)
Comment (facuspagnuolo) [reply packages/strategies/contracts/aave-v3/AaveV3Strategy.sol:93]: nope, cause `address[1]` generates a fixed-length memory array, this one is dynamic and it has to be that way in order to be compliant with the interface

File: packages/strategies/contracts/aave-v3/AaveV3Strategy.sol (hunk 2 above)
Comment (facuspagnuolo) [reply packages/strategies/contracts/aave-v3/AaveV3Strategy.sol:121]: we can add those here as well

**Pull request closed without merging**